docs/index.md

```diff
@@ -1 +1,293 @@
+---
+layout: ""
+page_title: "JFrog Xray Provider"
+description: |-
+  The Xray provider is used to interact with the resources supported by JFrog Xray.
+---
+
 # JFrog Xray Provider
+
+The [Xray](https://jfrog.com/xray/) provider is used to interact with the
+resources supported by JFrog Xray. Xray is a part of JFrog Artifactory and can't be used separately.
+The provider needs to be configured with the proper credentials before it can be used.
+Xray API documentation can be found [here](https://www.jfrog.com/confluence/display/JFROG/Xray+REST+API)
+
+Links to documentation for specific resources can be found in the table of contents to the left.
+
+This provider requires access to Artifactory and Xray APIs, which are only available in the _licensed_ pro and enterprise editions.
+You can determine which license you have by accessing the following URL
+`${host}/artifactory/api/system/licenses/`
+
+You can either access it via api, or web browser - it does require admin level credentials, but it's one of the few APIs that will work without a license (side node: you can also install your license here with a `POST`)
+
+```bash
+curl -sL ${host}/projects/api/system/licenses/ | jq .
+{
+  "type" : "Enterprise Plus Trial",
+  "validThrough" : "Jan 29, 2022",
+  "licensedTo" : "JFrog Ltd"
+}
+```
+
+The following 3 license types (`jq .type`) do **NOT** support APIs:
+- Community Edition for C/C++
+- JCR Edition
+- OSS
+
+## Example Usage
+
+```terraform
+# Required for Terraform 0.13 and up (https://www.terraform.io/upgrade-guides/0-13.html)
+terraform {
+  required_providers {
+    xray = {
+      source  = "registry.terraform.io/jfrog/xray"
+      version = "0.0.1"
+    }
+  }
+}
+
+provider "xray" {
+  // Supply the following env vars:
+  // ARTIFACTORY_URL (or JFROG_URL / PROJECTS_URL)
+  // XRAY_ACCESS_TOKEN or JFROG_ACCESS_TOKEN
+}
+
+resource "random_id" "randid" {
+  byte_length = 2
+}
+
+resource "xray_security_policy" "security1" {
```

Comment:
Let's name the resource with more descriptive name (vs just `security1`).

Reply:
I think in this file, we can provide just one example of policy and watch. The other examples should be in the `templates/resources/policy/resource.tf` per https://github.com/hashicorp/terraform-plugin-docs#conventional-paths

Reply:
I wanted to have all the functionality in one place. Is it not what we wanted?

Reply:
Not necessarily. When the doc is so long that it's not usable, we need to break it down to improve readability.

Reply:
done